projects/demo/src/app/breadcrumbs/breadcrumbs-routing.demo.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2016-2025 Broadcom. All Rights Reserved.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, NavigationEnd, Router } from '@angular/router';
+import { distinctUntilChanged, filter } from 'rxjs';
+
+import { MenuItem } from './breadcrumbs.demo.model';
+
+@Component({
+  selector: 'clr-breadcrumbs-routing-demo',
+  templateUrl: './breadcrumbs-routing.demo.html',
+})
+export class BreadcrumbsRoutingDemo implements OnInit {
+  menuItems = [];
+
+  constructor(private router: Router, private activatedRoute: ActivatedRoute) {}
+
+  ngOnInit(): void {
+    this.menuItems = this.createBreadcrumbs(this.activatedRoute.root);
+    console.log('Menu:', this.menuItems);
+    this.router.events
+      .pipe(
+        filter(event => event instanceof NavigationEnd),
+        distinctUntilChanged()
+      )
+      .subscribe(() => (this.menuItems = this.createBreadcrumbs(this.activatedRoute.root)));
```

Comment:
```suggestion
    this.menuItems = this.router.events.pipe(
      filter(event => event instanceof NavigationEnd),
      startWith(undefined),
      map(() => createBreadcrumbs(this.activatedRoute.root))
    );
```

The subscription is never unsubscribed. This can be simplified though so that we use the `async` pipe instead of an explicit subscription.

Reply:
See eb7e794f0106efa4cbb3b13b5ffe23753ca5b35f for an example refactor.

Reply:
Made changes are suggested in https://github.com/vmware-clarity/ng-clarity/commit/eb7e794f0106efa4cbb3b13b5ffe23753ca5b35f